src/client/diagnosticUtil.ts

```diff
@@ -132,14 +107,34 @@ export class DiagnosticUtil {
         const columnNumber = Number(matches[2]);
         diagnostic.lineNumber = lineNumber;
         diagnostic.columnNumber = columnNumber;
-        diagnostic.error = this.appendErrorWithLocation(diagnostic.error, lineNumber, columnNumber);
+        diagnostic.error = appendErrorWithLocation(diagnostic.error, lineNumber, columnNumber);
       }
     }
 
     return diagnostic;
   }
+}
+
+const appendErrorWithLocation = (error: string, line: string | number, column: string | number): string =>
```

Comment:
Man I'm all over using const & arrow funcs over named functions, but was curious about why you pulled the instance methods out into stand alone functions? Just a preference over static methods?

Reply:
it's part of the linter ruleset.  Methods that need to refer to class properties should be a method on the class.  Putting a private method that doesn't care about the class is unnecessary.

It has the obnoxious side effect of the TS compiler complaining about different implementations of private methods.  These are easier to refactor (keeping 100% internal) and more testable (export it and import it to your test, pass inputs, assert stuff vs. having to have a class instance to test).